File: aoc-2023/src/test/java/com/capital7software/aoc/aoc2023aoc/days/Day23Test.java
package com.capital7software.aoc.aoc2023aoc.days;

import static org.junit.jupiter.api.Assertions.assertEquals;

import com.capital7software.aoc.lib.AdventOfCodeTestBase;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

class Day23Test extends AdventOfCodeTestBase {
  private static final Logger LOGGER = LoggerFactory.getLogger(Day23Test.class);

  @BeforeEach
  void setUp() {
    var instance = new Day23();
    setupFromFile(instance.getDefaultInputFilename());
  }

  @Test
  void stepsOfLongestTrailWithSlopes() {
    var instance = new Day23();
    var expected = 94;
    var actual = instance.stepsOfLongestTrail(lines, false, true);
    assertEquals(expected, actual, "The number of steps "
        + "is not what was expected: " + expected);
  }

  @Test
  void stepsOfLongestTrailWithoutSlopes() {
    var instance = new Day23();
    var expected = 154;
    var actual = instance.stepsOfLongestTrail(lines, true, true);
    assertEquals(expected, actual, "The number of steps "
        + "is not what was expected: " + expected);
  }

  @Override
  protected Logger getLogger() {
    return LOGGER;
  }
}

File: aoc-2024/src/test/kotlin/com/capital7software/aoc/aoc2024aoc/days/Day23Test.kt
package com.capital7software.aoc.aoc2024aoc.days

import com.capital7software.aoc.lib.AdventOfCodeTestBase
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class Day23Test : AdventOfCodeTestBase() {
  companion object {
    val log: Logger = LoggerFactory.getLogger(Day23Test::class.java)
  }

  override fun getLogger(): Logger {
    return log
  }

  @BeforeEach
  fun setup() {
    val instance = Day23()
    setupFromFile(instance.defaultInputFilename)
  }

  @Test
  fun part1SampleShouldCountHistorianCandidateTriples() {
    val instance = Day23()
    val expected = 7
    val actual = instance.countHistorianCandidateTriples(lines)

    Assertions.assertEquals(
        expected,
        actual,
        "The number of historian candidate triples does not match the expected number."
    )
  }

  @Test
  fun part2SampleShouldFindLanPartyPassword() {
    val instance = Day23()
    val expected = "co,de,ka,ta"
    val actual = instance.lanPartyPassword(lines)

    Assertions.assertEquals(
        expected,
        actual,
        "The generated lan password does not match the expected password."
    )
  }
}
